control-plane/msp-operator/src/main.rs

```diff
@@ -435,75 +436,91 @@ impl ResourceContext {
+                        timeout: u32::pow(2, self.num_retries),
+                    });
+                }
 
-        // We are done creating the pool, we patched to created which triggers a
-        // new loop. Any error in the loop will call our error handler where we
-        // decide what to do
-        Ok(ReconcilerAction {
-            requeue_after: None,
-        })
+                let mut labels: HashMap<String, String> = HashMap::new();
+                labels.insert(
+                    String::from(constants::OPENEBS_CREATED_BY_KEY),
+                    String::from(constants::MSP_OPERATOR),
+                );
+
+                let body = CreatePoolBody::new_all(self.spec.disks.clone(), labels);
+                match self
+                    .pools_api()
+                    .put_node_pool(&self.spec.node, &self.name(), body)
+                    .await
+                {
+                    Ok(_) => {}
+                    Err(clients::tower::Error::Response(response))
+                        if response.status()
+                            == clients::tower::StatusCode::UNPROCESSABLE_ENTITY =>
+                    {
+                        // UNPROCESSABLE_ENTITY indicates that the pool spec already exists in the
+                        // control plane. So we want to update the CRD to
+                        // 'Created' to reflect this.
+                    }
+                    Err(e) => {
+                        return Err(e.into());
+                    }
+                };
+
+                self.k8s_notify(
+                    "Create or Import",
+                    "Created",
+                    "Created or imported pool",
+                    "Normal",
+                )
+                .await;
+
+                let _ = self.patch_status(MayastorPoolStatus::created()).await?;
+
+                // We are done creating the pool, we patched to created which triggers a
+                // new loop. Any error in the loop will call our error handler where we
+                // decide what to do
+                Ok(ReconcilerAction {
+                    requeue_after: None,
+                })
+            }
+            // We would land here if some error occurred ex, precondition failed, i.e. node
+            // down, in that case we check for pool existence before setting a status.
+            Err(_) => match self.pools_api().get_pool(&self.name()).await {
+                Ok(response) => {
+                    let pool = response.into_body();
+                    // As pool exists, set the status based on the presence of pool state.
+                    self.set_status_or_unknown(pool).await
+                }
+                Err(_) => {
+                    // If we don't find the pool, i.e. its not present or not yet created
+                    // so, set the status to Creating to retry creation.
+                    return self.is_missing().await;
```

Comment:
In this case should we not go into the error state? Otherwise, if we never created the pool and the node is not online, do we stay in creating forever?

Reply:
Yeah, we should move to Error, as if the node is not online and the pool didnot exist previously, there is no point in staying in Creating forever. Thanks.